pkg/snmp/traps/formatter.go

```diff
@@ -356,6 +361,20 @@ func formatType(variable gosnmp.SnmpPDU) string {
 func formatValue(variable gosnmp.SnmpPDU) interface{} {
 	switch variable.Value.(type) {
 	case []byte:
+		if variable.Type == gosnmp.IPAddress {
+			return common.IPBytesToString(variable.Value.([]byte))
```

Comment:
Normally, ipAdEntAddr / 1.3.6.1.2.1.4.20.1.1 in the example, is expected to be parsed by this gosnmp code and is expected to return a string: https://github.com/gosnmp/gosnmp/blob/master/helper.go#L124-L150
(But the screenshot indeed point to an issue with parsing and the value seems to be bytes).

Based on this code above from gosnmp, it's not possible to get value as []byte and type IPAddress (meaning that [this line might](https://github.com/DataDog/datadog-agent/pull/17743/files#diff-61a0f181efdcf8f6dcde9aaac3ebad9f5d5da92782138f3e681b31183397c9ddR365) never be reached).